Index: kernel-5_15/drivers/input/serio/libps2.c
===================================================================
--- kernel-5_15.orig/drivers/input/serio/libps2.c
+++ kernel-5_15/drivers/input/serio/libps2.c
@@ -188,6 +188,7 @@ bool ps2_is_keyboard_id(u8 id_byte)
 		0x5d,	/* Trust keyboard		*/
 		0x60,	/* NMB SGI keyboard, translated */
 		0x47,	/* NMB SGI keyboard		*/
+		0x83
 	};
 
 	return memchr(keyboard_ids, id_byte, sizeof(keyboard_ids)) != NULL;
