diff --git a/drivers/gpu/drm/drm_panel_orientation_quirks.c b/drivers/gpu/drm/drm_panel_orientation_quirks.c
index 58f5dc2f6dd5..f2bcf5f82b45 100644
--- a/drivers/gpu/drm/drm_panel_orientation_quirks.c
+++ b/drivers/gpu/drm/drm_panel_orientation_quirks.c
@@ -217,6 +217,12 @@ static const struct dmi_system_id orientation_data[] = {
 		  DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "LTH17"),
 		},
 		.driver_data = (void *)&lcd800x1280_rightside_up,
+  }, {
+    .matches = {
+      DMI_EXACT_MATCH(DMI_BIOS_VENDOR, "American Megatrends Inc."),
+      DMI_EXACT_MATCH(DMI_BIOS_VERSION, "SF133AR200_R11_Mipi"),
+    },
+    .driver_data = (void *)&asus_t100ha,
 	},
 	{}
 };
