diff --git a/src/android/camera_device.cpp b/src/android/camera_device.cpp
index 00d484716045..d1bfd1c1b7d5 100644
--- a/src/android/camera_device.cpp
+++ b/src/android/camera_device.cpp
@@ -304,7 +304,9 @@ int CameraDevice::initialize(const CameraConfigData *cameraConfigData)
 	 */
 	const ControlList &properties = camera_->properties();
 
-	if (properties.contains(properties::Location)) {
+	if (cameraConfigData && cameraConfigData->facing != -1) {
+		facing_ = cameraConfigData->facing;
+	} else if (properties.contains(properties::Location)) {
 		int32_t location = properties.get(properties::Location);
 		switch (location) {
 		case properties::CameraLocationFront:
@@ -335,13 +337,6 @@ int CameraDevice::initialize(const CameraConfigData *cameraConfigData)
 				<< "Camera location does not match"
 				<< " configuration file. Using " << facing_;
 		}
-	} else if (cameraConfigData) {
-		if (cameraConfigData->facing == -1) {
-			LOG(HAL, Error)
-				<< "Camera facing not in configuration file";
-			return -EINVAL;
-		}
-		facing_ = cameraConfigData->facing;
 	} else {
 		facing_ = CAMERA_FACING_EXTERNAL;
 	}
@@ -354,7 +349,9 @@ int CameraDevice::initialize(const CameraConfigData *cameraConfigData)
 	 * value for clockwise direction as required by the Android orientation
 	 * metadata.
 	 */
-	if (properties.contains(properties::Rotation)) {
+	if (cameraConfigData && cameraConfigData->rotation != -1) {
+		orientation_ = cameraConfigData->rotation;
+	} else if (properties.contains(properties::Rotation)) {
 		int rotation = properties.get(properties::Rotation);
 		orientation_ = (360 - rotation) % 360;
 		if (cameraConfigData && cameraConfigData->rotation != -1 &&
@@ -363,13 +360,6 @@ int CameraDevice::initialize(const CameraConfigData *cameraConfigData)
 				<< "Camera orientation does not match"
 				<< " configuration file. Using " << orientation_;
 		}
-	} else if (cameraConfigData) {
-		if (cameraConfigData->rotation == -1) {
-			LOG(HAL, Error)
-				<< "Camera rotation not in configuration file";
-			return -EINVAL;
-		}
-		orientation_ = cameraConfigData->rotation;
 	} else {
 		orientation_ = 0;
 	}
