--- a/trunks/BUILD.gn
+++ b/trunks/BUILD.gn
@@ -53,6 +53,9 @@
     "BIOMETRICS_DEV=true",
   ]
   libs = [ "hwsec-foundation" ]
+  if (use.tpm2_simulator) {
+    defines += [ "USE_SIMULATOR=1" ]
+  }
 }
 
 proto_library("interface_proto") {
@@ -191,6 +194,11 @@
     "trunks_dbus_service.cc",
     "trunks_metrics.cc",
   ]
+
+  if (use.tpm2_simulator) {
+    sources += [ "tpm_simulator_handle.cc" ]
+  }
+
   all_dependent_configs = [
     ":trunksd_lib_all_dependent_configs",
     ":trunksd_lib_all_dependent_pkg_deps",
@@ -222,6 +230,9 @@
     ":trunks",
     ":trunksd_lib",
   ]
+  if (use.tpm2_simulator) {
+    libs = [ "tpm2" ]
+  }
 }
 
 executable("tpm_tunneld") {
--- a/trunks/trunksd.cc
+++ b/trunks/trunksd.cc
@@ -21,6 +21,9 @@
 #include "trunks/power_manager.h"
 #include "trunks/resource_manager.h"
 #include "trunks/tpm_handle.h"
+#if defined(USE_SIMULATOR)
+#include "trunks/tpm_simulator_handle.h"
+#endif
 #include "trunks/trunks_dbus_service.h"
 #include "trunks/trunks_factory_impl.h"
 #include "trunks/trunks_ftdi_spi.h"
@@ -98,6 +101,11 @@
   if (cl->HasSwitch("ftdi")) {
     LOG(INFO) << "Sending commands to FTDI SPI.";
     low_level_transceiver = new trunks::TrunksFtdiSpi();
+#if defined(USE_SIMULATOR)
+  } else if (cl->HasSwitch("simulator")) {
+    LOG(INFO) << "Sending commands to simulator.";
+    low_level_transceiver = new trunks::TpmSimulatorHandle();
+#endif
   } else {
     low_level_transceiver = new trunks::TpmHandle();
   }
