diff --git a/trunks/BUILD.gn b/trunks/BUILD.gn
index 7556359a72..0a1be02ac0 100644
--- a/trunks/BUILD.gn
+++ b/trunks/BUILD.gn
@@ -62,6 +62,9 @@ config("target_defaults") {
     "BIOMETRICS_DEV=true",
   ]
   libs = [ "hwsec-foundation" ]
+  if (use.tpm2_simulator) {
+    defines += [ "USE_SIMULATOR=1" ]
+  }
 }
 
 proto_library("interface_proto") {
@@ -203,6 +206,11 @@ static_library("trunksd_lib") {
     "trunks_dbus_service.cc",
     "trunks_metrics.cc",
   ]
+
+  if (use.tpm2_simulator) {
+    sources += [ "tpm_simulator_handle.cc" ]
+  }
+
   all_dependent_configs = [
     ":trunksd_lib_all_dependent_configs",
     ":trunksd_lib_all_dependent_pkg_deps",
@@ -235,6 +243,9 @@ executable("trunksd") {
     ":trunks",
     ":trunksd_lib",
   ]
+  if (use.tpm2_simulator) {
+    libs += [ "tpm2" ]
+  }
 }
 
 executable("tpm_tunneld") {
diff --git a/trunks/trunksd.cc b/trunks/trunksd.cc
index dee990ea25..2d7f61f769 100644
--- a/trunks/trunksd.cc
+++ b/trunks/trunksd.cc
@@ -24,6 +24,9 @@
 #include "trunks/resilience/write_error_tracker_impl.h"
 #include "trunks/resource_manager.h"
 #include "trunks/tpm_handle.h"
+#if defined(USE_SIMULATOR)
+#include "trunks/tpm_simulator_handle.h"
+#endif
 #include "trunks/trunks_dbus_service.h"
 #include "trunks/trunks_factory_impl.h"
 #include "trunks/trunks_ftdi_spi.h"
@@ -105,6 +108,11 @@ int main(int argc, char** argv) {
   if (cl->HasSwitch("ftdi")) {
     LOG(INFO) << "Sending commands to FTDI SPI.";
     low_level_transceiver = new trunks::TrunksFtdiSpi();
+#if defined(USE_SIMULATOR)
+  } else if (cl->HasSwitch("simulator")) {
+    LOG(INFO) << "Sending commands to simulator.";
+    low_level_transceiver = new trunks::TpmSimulatorHandle();
+#endif
   } else {
     low_level_transceiver = new trunks::TpmHandle(write_error_tracker);
   }
