Index: kernel-5_4/drivers/net/wireless/intel/iwlwifi/cfg/22000.c
===================================================================
--- kernel-5_4.orig/drivers/net/wireless/intel/iwlwifi/cfg/22000.c
+++ kernel-5_4/drivers/net/wireless/intel/iwlwifi/cfg/22000.c
@@ -56,7 +56,7 @@
 #include "iwl-config.h"
 
 /* Highest firmware API version supported */
-#define IWL_22000_UCODE_API_MAX	50
+#define IWL_22000_UCODE_API_MAX	59
 
 /* Lowest firmware API version supported */
 #define IWL_22000_UCODE_API_MIN	39
