Index: kernel-6_6/scripts/setlocalversion
===================================================================
--- kernel-6_6.orig/scripts/setlocalversion
+++ kernel-6_6/scripts/setlocalversion
@@ -92,7 +92,7 @@ scm_version()
 		# If only the short version is requested, don't bother
 		# running further git commands
 		if $short; then
-			echo "+"
+			echo ""
 			return
 		fi
 		# If we are past the tagged commit, we pretty print it.
