Index: v6.6/security/chromiumos/lsm.c
===================================================================
--- v6.6.orig/security/chromiumos/lsm.c
+++ v6.6/security/chromiumos/lsm.c
@@ -84,6 +84,9 @@ static void report(const char *origin, c
 	kfree(cmdline);
 	kfree(alloced);
 }
+#else
+static void report(const char *origin, const struct path *path, char *operation)
+{ }
 #endif
 
 static int chromiumos_security_sb_mount(const char *dev_name,
