From: Fangzhou Chen <fangzhou@fydeos.io>
Date: Fri, 23 Dec 2022 04:39:01 +0000
Subject: Prefer to use navigator.language over navigator.languages
Reference: https://fydeos.atlassian.net/browse/OS-198
Signed-off-by: Fangzhou Chen <fangzhou@fydeos.io>

--- a/libdot/js/lib_i18n.js
+++ b/libdot/js/lib_i18n.js
@@ -39,6 +39,8 @@
 lib.i18n.getAcceptLanguages = function() {
   if (lib.i18n.browser_) {
     return new Promise((resolve) => {
+      const lang = typeof lib.i18n.browser_.getUILanguage === 'function' ?
+                   lib.i18n.browser_.getUILanguage() :'';
       lib.i18n.browser_.getAcceptLanguages((languages) => {
         // Chrome might be in a bad state and not return any languages.  If we
         // pass this up to the caller who isn't expecting undefined, they'll
@@ -50,12 +52,11 @@
           languages = ['en'];
         }
 
-        resolve(languages);
+        resolve(lang ? [lang, ...languages] : languages);
       });
     });
   } else {
-    const languages = navigator.languages || [navigator.language];
-    return Promise.resolve(languages);
+    return Promise.resolve([navigator.language, ...navigator.languages]);
   }
 };
 
