diff --git a/mems_setup/sensor_kind.cc b/mems_setup/sensor_kind.cc
index 1360e7b028..e69e649c47 100644
--- a/mems_setup/sensor_kind.cc
+++ b/mems_setup/sensor_kind.cc
@@ -28,7 +28,7 @@ constexpr char kProxDeviceNames[][23] = {
     "sx9324",           "sx932x",       "sx9360", "cros-ec-mkbp-proximity"};
 
 constexpr char kHidDeviceNames[][13] = {
-    "accel_3d",    "gyro_3d",  "magn_3d",     "als",
+    "accel_3d",    "gyro_3d",  "magn_3d",     "als", "lis2dw12",
     "temperature", "incli_3d", "dev_rotation"};
 }  // namespace
 
