Index: shill/cellular/cellular_capability_3gpp.cc
===================================================================
--- shill.orig/cellular/cellular_capability_3gpp.cc
+++ shill/cellular/cellular_capability_3gpp.cc
@@ -1809,6 +1809,16 @@ void CellularCapability3gpp::OnModemProp
       mdn = numbers[0];
     OnMdnChanged(mdn);
   }
+
+  if (properties.ContainsVariant(MM_MODEM_PROPERTY_SIGNALQUALITY)) {
+    uint32_t signal;
+    bool enabled;
+    std::tie(signal, enabled) = properties.GetVariant(MM_MODEM_PROPERTY_SIGNALQUALITY).Get<SignalQuality>();
+    SLOG(this, 1) << __func__ << " signal:" << signal << " enabled:" << enabled;
+    if (enabled) {
+      cellular()->HandleNewSignalQuality(signal);
+    }
+  }
 }
 
 void CellularCapability3gpp::OnPropertiesChanged(
