--- a/oobe_config/etc/init/oobe_config_restore.conf
+++ b/oobe_config/etc/init/oobe_config_restore.conf
@@ -38,8 +38,8 @@
   # fail. Use $@ array instead of empty string, as passing in "" to minijail
   # also fails.
   set --
-  if [ -d /mnt/stateful_partition/unencrypted/flex_config ]; then
-    set -- "$@" -b /mnt/stateful_partition/unencrypted/flex_config,,1
+  if [ -d /usr/share/oem/flex_config ]; then
+    set -- "$@" -b /usr/share/oem/flex_config,,1
   fi
 
   exec minijail0 --config=/usr/share/minijail/oobe_config_restore.conf \
--- a/oobe_config/filesystem/file_handler.h
+++ b/oobe_config/filesystem/file_handler.h
@@ -193,7 +193,7 @@
   static constexpr char kFlexOobeConfigUnencryptedDirPath[] =
       "mnt/stateful_partition/unencrypted/flex_config";
   static constexpr char kFlexOobeConfigUnencryptedFilePath[] =
-      "mnt/stateful_partition/unencrypted/flex_config/config.json";
+      "usr/share/oem/flex_config/config.json";
   static constexpr char kFlexOobeConfigEncryptedDirPath[] =
       "var/lib/oobe_config_restore/flex_config";
   static constexpr char kFlexOobeConfigEncryptedFilePath[] =
--- a/oobe_config/minijail/oobe_config_restore.conf
+++ b/oobe_config/minijail/oobe_config_restore.conf
@@ -44,6 +44,7 @@
 bind-mount = /var/lib/metrics/structured/events,,1
 bind-mount = /var/lib/oobe_config_restore,,1
 mount = /mnt/stateful_partition,/mnt/stateful_partition,tmpfs,MS_NODEV|MS_NOEXEC|MS_NOSUID,mode=755,size=10M
+mount = /usr/share/oem,/usr/share/oem,tmpfs,MS_NODEV|MS_NOEXEC|MS_NOSUID,mode=755,size=10M
 bind-mount = /mnt/stateful_partition/unencrypted/preserve,,1
 u = oobe_config_restore
 G
@@ -55,7 +56,7 @@
 fs-path-rw = /var/lib/metrics/structured/events
 fs-path-rw = /var/lib/oobe_config_restore
 fs-path-rw = /mnt/stateful_partition/unencrypted/preserve
-fs-path-rw = /mnt/stateful_partition/unencrypted/flex_config
+fs-path-rw = /usr/share/oem/flex_config
 fs-path-ro = /dev/log
 fs-path-ro = /sys/fs/pstore
 fs-path-ro = /sys/block
