diff --git a/attestation/common/tpm_utility_common.cc b/attestation/common/tpm_utility_common.cc
index c898981b32..fa8673cf2f 100644
--- a/attestation/common/tpm_utility_common.cc
+++ b/attestation/common/tpm_utility_common.cc
@@ -52,7 +52,7 @@ void TpmUtilityCommon::UpdateTpmStatus() {
 
   if (!tpm_manager_utility_->GetTpmStatus(&is_enabled, &is_owned,
                                           &local_data)) {
-    LOG(ERROR) << __func__ << ": Failed to get tpm status from tpm_manager.";
+    VLOG(1) << __func__ << ": Failed to get tpm status from tpm_manager.";
     return;
   }
 
