Index: crosh/src/dev/shell.rs
===================================================================
--- crosh.orig/src/dev/shell.rs
+++ crosh/src/dev/shell.rs
@@ -96,14 +96,7 @@ fn execute_shell(_cmd: &Command, args: &
     } else {
         if is_no_new_privs_set() {
             println!(
-                r#"Sudo commands will not succeed by default.
-If you want to use sudo commands, use the VT-2 shell
-(Ctrl-Alt-{{F2/Right arrow/Refresh}}) or build the image with the
-login_enable_crosh_sudo USE flag:
-
-$ USE=login_enable_crosh_sudo emerge-$BOARD chromeos-login
-or
-$ USE=login_enable_crosh_sudo cros build-packages --board=$BOARD
+                r#"
     "#
             );
         }
