diff --git a/crosh/src/dev/shell.rs b/crosh/src/dev/shell.rs
index 048b1da113..5f2c1c3c6d 100644
--- a/crosh/src/dev/shell.rs
+++ b/crosh/src/dev/shell.rs
@@ -25,17 +25,6 @@ pub fn register(dispatcher: &mut Dispatcher) {
 }
 
 fn execute_shell(_cmd: &Command, _args: &Arguments) -> Result<(), dispatcher::Error> {
-    println!(
-        r#"Sudo commands will not succeed by default.
-If you want to use sudo commands, use the VT-2 shell
-(Ctrl-Alt-{{F2/Right arrow/Refresh}}) or build the image with the
-login_enable_crosh_sudo USE flag:
-
-$ USE=login_enable_crosh_sudo emerge-$BOARD chromeos-login
-or
-$ USE=login_enable_crosh_sudo cros build-packages --board=$BOARD
-    "#
-    );
     wait_for_result(
         process::Command::new(get_shell())
             .arg("-l")
