diff --git a/installer/chromeos-install.sh b/installer/chromeos-install.sh
old mode 100755
new mode 100644
index 247fe18b6b..61ea71a6eb
--- a/installer/chromeos-install.sh
+++ b/installer/chromeos-install.sh
@@ -903,6 +903,9 @@ main() {
   # get installed last. The order shouldn't matter but legacy behavior has
   # us go in reverse order.
 
+  # Flush filesystem buffers and clear page cache, dentries, and inodes
+  sync && echo 3 > /proc/sys/vm/drop_caches
+
   # 12
   copy_partition "${PARTITION_NUM_EFI_SYSTEM:?}" "${SRC}" "${DST}" 1 1 false
 
