Index: patchpanel/arc_service.cc
===================================================================
--- patchpanel.orig/arc_service.cc
+++ patchpanel/arc_service.cc
@@ -527,7 +527,7 @@ void ArcService::AddDevice(const ShillCl
   RecordEvent(metrics_, ArcServiceUmaEvent::kAddDevice);
 
   if (devices_.find(shill_device.ifname) != devices_.end()) {
-    LOG(DFATAL) << shill_device.logging_tag << " " << __func__
+    LOG(WARNING) << shill_device.logging_tag << " " << __func__
                 << ": Attemping to add already tracked shill device "
                 << shill_device;
     return;
