Index: patchpanel/arc_service.cc
===================================================================
--- patchpanel.orig/arc_service.cc
+++ patchpanel/arc_service.cc
@@ -695,7 +695,7 @@ void ArcService::AddDevice(const ShillCl
   RecordEvent(metrics_, ArcServiceUmaEvent::kAddDevice);
 
   if (devices_.find(shill_device.ifname) != devices_.end()) {
-    LOG(DFATAL) << "Attemping to add already tracked shill device "
+    LOG(WARNING) << "Attemping to add already tracked shill device "
                 << shill_device;
     return;
   }
