diff --git a/chromeos-config/scripts/cros_config_setup_legacy.sh b/chromeos-config/scripts/cros_config_setup_legacy.sh
index 8a1fc592eb..4797582059 100755
--- a/chromeos-config/scripts/cros_config_setup_legacy.sh
+++ b/chromeos-config/scripts/cros_config_setup_legacy.sh
@@ -33,19 +33,8 @@ setconfig() {
     fi
 }
 
-model="$(mosys platform model)"
-brand_code="$(mosys platform brand)"
-customization_id="$(mosys platform customization || true)"
-platform_name="$(mosys platform name)"
-
-setconfig / brand-code "${brand_code}"
-setconfig / name "${model}"
-setconfig /firmware image-name "${model}"
 setconfig /hardware-properties form-factor CHROMEBOOK
 setconfig /hardware-properties has-backlight true
-setconfig /hardware-properties psu-type battery
-setconfig /identity platform-name "${platform_name}"
-setconfig /ui help-content-id "${customization_id}"
 
 mkdir -p "${MOUNTPOINT}/v1"
 mount -n -obind,ro,nodev,noexec,nosuid "${CONFIG_OUT}" "${MOUNTPOINT}/v1"
