index e30c0c35b7..95513be33f 100644
Index: init/BUILD.gn
===================================================================
--- init.orig/BUILD.gn
+++ init/BUILD.gn
@@ -74,6 +74,7 @@ pkg_config("target_defaults") {
     "USE_TPM1=${use.tpm}",
     "USE_TPM2=${use.tpm2}",
     "USE_TPM2_SIMULATOR=${use.tpm2_simulator}",
+    "USE_TPM2_SIMULATOR_DEPRECATED=${use.tpm2_simulator_deprecated}",
     "USE_VTPM_PROXY=${use.vtpm_proxy}",
   ]
   pkg_deps = [
Index: init/tpm_encryption/tpm_setup.cc
===================================================================
--- init.orig/tpm_encryption/tpm_setup.cc
+++ init/tpm_encryption/tpm_setup.cc
@@ -282,7 +282,7 @@ bool TpmSystemKey::ShallUseTpmForSystemK
   }
 
   /* Assume we have tpm for system_key when we are using vtpm tpm2 simulator. */
-  return USE_TPM2_SIMULATOR && USE_VTPM_PROXY;
+  return USE_TPM2_SIMULATOR && USE_VTPM_PROXY && !USE_TPM2_SIMULATOR_DEPRECATED;
 }
 
 bool TpmSystemKey::MigrateTpmOwnerShipStateFile() {
