diff --git a/security/selinux/hooks.c b/security/selinux/hooks.c
index 07d000bda174..caef3e23e612 100644
--- a/security/selinux/hooks.c
+++ b/security/selinux/hooks.c
@@ -93,6 +93,10 @@
 #include <linux/fanotify.h>
 #include <linux/io_uring.h>
 
+#ifdef CONFIG_OVERLAY_FS
+#include "../../fs/overlayfs/ovl_entry.h"
+#endif
+
 #include "avc.h"
 #include "objsec.h"
 #include "netif.h"
@@ -104,6 +108,30 @@
 #include "audit.h"
 #include "avc_ss.h"
 
+static bool is_overlay_sb(struct super_block *sb)
+{
+ const char* fstype = sb->s_type->name;
+ return strcmp(fstype, "overlay") == 0;
+}
+
+static bool is_overlay_inode(struct inode *inode)
+{
+ return is_overlay_sb(inode->i_sb);
+}
+
+#ifdef CONFIG_OVERLAY_FS
+
+static struct inode *get_real_inode_from_ovl(struct inode *inode) {
+ struct ovl_inode *oi = OVL_I(inode);
+ struct dentry *upperdentry = ovl_upperdentry_dereference(oi);
+ struct dentry *lowerdentry = oi->lowerpath.dentry;
+
+ 
+ return upperdentry ? d_inode(upperdentry) :
+	 (lowerdentry ? d_inode(lowerdentry) : NULL);
+}
+#endif
+
 struct selinux_state selinux_state;
 
 /* SECMARK reference count */
@@ -456,6 +484,10 @@ static int selinux_is_sblabel_mnt(struct super_block *sb)
 	 */
 	BUILD_BUG_ON(SECURITY_FS_USE_MAX != 7);
 
+#ifdef CONFIG_OVERLAY_FS
+	if(is_overlay_sb(sb))
+		return 1;
+#endif
 	switch (sbsec->behavior) {
 	case SECURITY_FS_USE_XATTR:
 	case SECURITY_FS_USE_TRANS:
@@ -3396,7 +3428,16 @@ static int selinux_inode_getsecurity(struct user_namespace *mnt_userns,
 	 * and lack of permission just means that we fall back to the
 	 * in-core context value, not a denial.
 	 */
-	isec = inode_security(inode);
+	if (is_overlay_inode(inode)) {
+#ifdef CONFIG_OVERLAY_FS
+		isec = inode_security(get_real_inode_from_ovl(inode));
+#else
+		isec = inode_security(inode);
+#endif
+	}else {
+		isec = inode_security(inode);
+	}
+
 	if (has_cap_mac_admin(false))
 		error = security_sid_to_context_force(&selinux_state,
 						      isec->sid, &context,
