Index: vm_tools/concierge/untrusted_vm_utils.cc
===================================================================
--- vm_tools.orig/concierge/untrusted_vm_utils.cc
+++ vm_tools/concierge/untrusted_vm_utils.cc
@@ -181,7 +181,7 @@ bool UntrustedVMUtils::SafeToRunVirtualM
     // l1tf and mds mitigations then fail to start an untrusted VM.
     case UntrustedVMUtils::MitigationStatus::VULNERABLE:
       *reason = "Host vulnerable against untrusted VM";
-      return false;
+      return true;
 
     // At this point SMT should not be a security issue, since Linux 4.19+ has
     // security patches to make nested VMs co-exist securely with SMT.
