Index: vm_tools/concierge/untrusted_vm_utils.cc
===================================================================
--- vm_tools.orig/concierge/untrusted_vm_utils.cc
+++ vm_tools/concierge/untrusted_vm_utils.cc
@@ -189,6 +189,10 @@ UntrustedVMUtils::CheckUntrustedVMMitiga
 
 bool UntrustedVMUtils::SafeToRunVirtualMachines(std::string* reason) const {
   DCHECK(reason);
+  const bool force_enabled = true;
+  if (force_enabled) {
+    return true;
+  }
 
   // Check if l1tf and mds mitigations are present on the host.
   switch (CheckUntrustedVMMitigationStatus()) {
