Index: vm_tools/concierge/service.cc
===================================================================
--- vm_tools.orig/concierge/service.cc
+++ vm_tools/concierge/service.cc
@@ -208,7 +208,7 @@ constexpr base::TimeDelta kDiskOpReportI
 // The minimum kernel version of the host which supports untrusted VMs or a
 // trusted VM with nested VM support.
 constexpr KernelVersionAndMajorRevision
-    kMinKernelVersionForUntrustedAndNestedVM = std::make_pair(4, 19);
+    kMinKernelVersionForUntrustedAndNestedVM = std::make_pair(5, 19);
 
 // The minimum kernel version of the host which supports virtio-pmem.
 constexpr KernelVersionAndMajorRevision kMinKernelVersionForVirtioPmem =
@@ -431,6 +431,8 @@ void SetVmCpuArgs(int32_t cpus, VmBuilde
 
   if (!vm_cpu_args.cpu_clusters.empty()) {
     for (const auto& cluster : vm_cpu_args.cpu_clusters) {
+      if (cluster.empty())
+        continue;
       auto cpu_list = base::JoinString(cluster, ",");
       vm_builder.AppendCustomParam("--cpu-cluster", cpu_list);
     }
