diff --git a/vm_tools/concierge/service.cc b/vm_tools/concierge/service.cc
index 86ab00c7b..05772f37c 100644
--- a/vm_tools/concierge/service.cc
+++ b/vm_tools/concierge/service.cc
@@ -146,7 +146,7 @@ constexpr base::TimeDelta kDiskOpReportInterval =
 // The minimum kernel version of the host which supports untrusted VMs or a
 // trusted VM with nested VM support.
 constexpr KernelVersionAndMajorRevision
-    kMinKernelVersionForUntrustedAndNestedVM = std::make_pair(4, 19);
+    kMinKernelVersionForUntrustedAndNestedVM = std::make_pair(5, 19);
 
 // The minimum kernel version of the host which supports virtio-pmem.
 constexpr KernelVersionAndMajorRevision kMinKernelVersionForVirtioPmem =
