Index: vm_tools/concierge/service_arc.cc
===================================================================
--- vm_tools.orig/concierge/service_arc.cc
+++ vm_tools/concierge/service_arc.cc
@@ -91,9 +91,10 @@ const VariationsFeature kArcVmAAudioMMAP
 // file when in dev mode.
 base::FilePath GetImagePath(const base::FilePath& image_path,
                             bool is_dev_mode) {
+#if 0
   if (!is_dev_mode)
     return image_path;
-
+#endif
   // When in dev mode, the Android images might be on the stateful partition and
   // |kRootfsPath| might be a symlink to the stateful partition image file. In
   // that case, we need to use the resolved path so that brillo::SafeFD calls
