diff --git a/vm_tools/concierge/service.cc b/vm_tools/concierge/service.cc
index 2808c13ab..02df98628 100644
--- a/vm_tools/concierge/service.cc
+++ b/vm_tools/concierge/service.cc
@@ -75,6 +75,10 @@ constexpr char kVmRootfsName[] = "vm_rootfs.img";
 
 // Maximum number of extra disks to be mounted inside the VM.
 constexpr int kMaxExtraDisks = 10;
+//---***FYDEOS BEGIN***---
+constexpr int kMountStateFsTimeoutSeconds = 30;
+constexpr int kRunLxdScriptTimeoutSeconds = 20;
+//---***FYDEOS END***---
 
 // How long to wait before timing out on `lxd waitready`.
 constexpr int kLxdWaitreadyTimeoutSeconds = 50;
@@ -1146,7 +1150,8 @@ bool Service::StartTermina(VirtualMachine* vm, string* failure_reason) {
 
   // Set up the stateful disk. This will format the disk if necessary, then
   // mount it.
-  if (!vm->RunProcess({"stateful_setup.sh"}, kLxdEnv)) {
+  if (!vm->RunProcessWithTimeout({"stateful_setup.sh"}, kLxdEnv,
+                        base::TimeDelta::FromSeconds(kMountStateFsTimeoutSeconds))) {
     LOG(ERROR) << "Stateful setup failed";
     *failure_reason = "stateful setup failed";
     return false;
@@ -1172,7 +1177,8 @@ bool Service::StartTermina(VirtualMachine* vm, string* failure_reason) {
 
   // Perform any setup for lxd to be usable. On first run, this sets up the
   // lxd configuration (network bridge, storage pool, etc).
-  if (!vm->RunProcess({"lxd_setup.sh"}, kLxdEnv)) {
+  if (!vm->RunProcessWithTimeout({"lxd_setup.sh"}, kLxdEnv,
+                        base::TimeDelta::FromSeconds(kRunLxdScriptTimeoutSeconds))) {
     LOG(ERROR) << "lxd setup failed";
     *failure_reason = "lxd setup failed";
     return false;
