Index: login_manager/chrome_setup.cc
===================================================================
--- login_manager.orig/chrome_setup.cc
+++ login_manager/chrome_setup.cc
@@ -692,6 +692,36 @@ void SetUpRegulatoryLabelFlag(ChromiumCo
   }
 }
 
+bool AddDynamicDefaultWallpaperFlags(
+    ChromiumCommandBuilder* builder,
+    const base::RepeatingCallback<bool(const base::FilePath&)>& path_exists) {
+  const std::string base_path = "/usr/share/chromeos-assets/wallpaper";
+  std::vector<std::string> filenames = {
+      "default_large", "default_large.dark", "default_large.light",
+      "default_small", "default_small.dark", "default_small.light",
+  };
+  std::vector<std::string> filetypes = { "jpg", "png" };
+  bool support = false;
+  for (const auto& filetype : filetypes) {
+    support = true;
+    for (const auto& filename : filenames) {
+      base::FilePath path(base::StringPrintf("%s/%s.%s",
+                                             base_path.c_str(),
+                                             filename.c_str(),
+                                             filetype.c_str()));
+      if (!path_exists.Run(path)) {
+        support = false;
+        break;
+      }
+    }
+    if (support) {
+      builder->AddArg("--fyde-dynamic-default-wallpaper");
+      break;
+    }
+  }
+  return support;
+}
+
 void SetUpWallpaperFlags(
     ChromiumCommandBuilder* builder,
     brillo::CrosConfigInterface* cros_config,
@@ -718,6 +748,7 @@ void SetUpWallpaperFlags(
 
   // Fall back to default.
   AddWallpaperFlags(builder, "default", "default", path_exists);
+  AddDynamicDefaultWallpaperFlags(builder, path_exists);
 }
 
 void SetUpInternalStylusFlag(ChromiumCommandBuilder* builder,
