diff --git a/missive/storage/storage_module.cc b/missive/storage/storage_module.cc
index 57f0d3ebef..16174bf15e 100644
--- a/missive/storage/storage_module.cc
+++ b/missive/storage/storage_module.cc
@@ -87,7 +87,7 @@ void StorageModule::Create(
 
   // TOOD(b/279057326): dynamically update storage implementation when
   // `legacy_storage_enabled` changes
-  if (instance->legacy_storage_enabled()) {
+  if (legacy_storage_enabled) {
     Storage::Create(options, async_start_upload_cb, queues_container,
                     encryption_module, compression_module,
                     std::move(completion_cb));
