Index: update_engine/BUILD.gn
===================================================================
--- update_engine.orig/BUILD.gn
+++ update_engine/BUILD.gn
@@ -197,6 +197,8 @@ if (!use.cros_host) {
       "cros/real_system_state.cc",
       "cros/shill_proxy.cc",
       "cros/update_attempter.cc",
+      "cros/fydeos_license_checker.cc",
+      "cros/fydeos_toggle_ota.cc",
       "libcurl_http_fetcher.cc",
       "metrics_utils.cc",
       "update_boot_flags_action.cc",
Index: update_engine/cros/daemon_chromeos.cc
===================================================================
--- update_engine.orig/cros/daemon_chromeos.cc
+++ update_engine/cros/daemon_chromeos.cc
@@ -63,6 +63,8 @@ void DaemonChromeOS::OnDBusRegistered(bo
     return;
   }
 
+  license_checker_.Setup();
+
   // Update the telemetry information before starting the updater, to request
   // once and continue caching on boot.
   SystemState::Get()->cros_healthd()->ProbeTelemetryInfo(
@@ -74,7 +76,7 @@ void DaemonChromeOS::OnDBusRegistered(bo
           TelemetryCategoryEnum::kBus,
       },
       base::BindOnce(
-          []() { SystemState::Get()->update_attempter()->StartUpdater(); }));
+          []() { fydeos::FydeLicenseChecker::Get().Init(); SystemState::Get()->update_attempter()->StartUpdater(); }));
 }
 
 }  // namespace chromeos_update_engine
Index: update_engine/cros/daemon_chromeos.h
===================================================================
--- update_engine.orig/cros/daemon_chromeos.h
+++ update_engine/cros/daemon_chromeos.h
@@ -12,6 +12,9 @@
 #include "update_engine/common/subprocess.h"
 #include "update_engine/cros/dbus_service.h"
 #include "update_engine/cros/real_system_state.h"
+// ---***FYDEOS BEGIN***---
+#include "update_engine/cros/fydeos_license_checker.h"
+// ---***FYDEOS END***---
 
 namespace chromeos_update_engine {
 
@@ -36,6 +39,8 @@ class DaemonChromeOS : public DaemonBase
   // dependency.
   Subprocess subprocess_;
 
+  fydeos::FydeLicenseChecker license_checker_;
+
   // |SystemState| is a global context, but we can't have a static singleton of
   // its object because the style guide does not allow that (it has non-trivial
   // dtor). We need an instance of |SystemState| in this class instead and have
Index: update_engine/cros/fydeos_license_checker.cc
===================================================================
--- /dev/null
+++ update_engine/cros/fydeos_license_checker.cc
@@ -0,0 +1,47 @@
+#include "update_engine/cros/fydeos_license_checker.h"
+
+#include <base/logging.h>
+#include <base/strings/stringprintf.h>
+#include <base/json/json_reader.h>
+#include <base/base64.h>
+#include <base/values.h>
+#include <crypto/signature_verifier.h>
+
+#include "update_engine/common/subprocess.h"
+
+namespace fydeos {
+
+namespace {
+  const char kShellCmd[] = "/usr/share/fydeos_shell/license-utils.sh";
+  const char kParamId[] = "id";
+} // namespace
+
+FydeLicenseChecker::FydeLicenseChecker() = default;
+FydeLicenseChecker::~FydeLicenseChecker() {
+  if (license_checker_singleton_ == this) {
+    license_checker_singleton_ = nullptr;
+  }
+}
+
+void FydeLicenseChecker::Setup() {
+  if (license_checker_singleton_ == this)
+    return;
+  CHECK(license_checker_singleton_ == nullptr);
+  license_checker_singleton_ = this;
+}
+
+void FydeLicenseChecker::Init() {
+  int exit_code = 0;
+  std::string value, error;
+  std::vector<std::string> cmd = {kShellCmd, kParamId};
+  if (!chromeos_update_engine::Subprocess::SynchronousExec(cmd, &exit_code, &value, &error) || exit_code) {
+    LOG(ERROR) << "Get fydeos license id error, exit_code: " << exit_code;
+    return;
+  }
+  base::TrimWhitespaceASCII(value, base::TRIM_ALL, &license_id_);
+  LOG(INFO) << "FydeOS license id: " << license_id_;
+}
+
+FydeLicenseChecker* FydeLicenseChecker::license_checker_singleton_ = nullptr;
+
+} // namespace fydeos
Index: update_engine/cros/fydeos_license_checker.h
===================================================================
--- /dev/null
+++ update_engine/cros/fydeos_license_checker.h
@@ -0,0 +1,26 @@
+#ifndef UPDATE_ENGINE_FYDEOS_LICENSE_CHECKER_H
+#define UPDATE_ENGINE_FYDEOS_LICENSE_CHECKER_H
+
+#include <string>
+#include <memory>
+
+namespace fydeos {
+
+class FydeLicenseChecker {
+  public:
+    explicit FydeLicenseChecker();
+    ~FydeLicenseChecker();
+
+    static FydeLicenseChecker& Get() { return *license_checker_singleton_; }
+    void Setup();
+    void Init();
+    std::string GetLicenseId() const { return license_id_; };
+
+  private:
+    static FydeLicenseChecker* license_checker_singleton_;
+
+    std::string license_id_;
+};
+
+} // namespace fydeos
+#endif /* ifndef UPDATE_ENGINE_FYDEOS_LICENSE_CHECKER_H */
Index: update_engine/cros/fydeos_toggle_ota.cc
===================================================================
--- /dev/null
+++ update_engine/cros/fydeos_toggle_ota.cc
@@ -0,0 +1,15 @@
+#include "update_engine/cros/fydeos_toggle_ota.h"
+#include <base/files/file_util.h>
+
+namespace fydeos {
+namespace {
+  // keep it the same with chromium src fydeos/misc/fydeos_toggle_ota.cc
+  const char kFydeOSOTAIndicatorFile[] = "/mnt/stateful_partition/unencrypted/preserve/.disable_fydeos_ota";
+}
+
+bool OTAEnabled() {
+  const bool enabled = !base::PathExists(base::FilePath(kFydeOSOTAIndicatorFile));
+  return enabled;
+}
+
+} // fydeos
Index: update_engine/cros/fydeos_toggle_ota.h
===================================================================
--- /dev/null
+++ update_engine/cros/fydeos_toggle_ota.h
@@ -0,0 +1,8 @@
+#ifndef FYDEOS_TOGGLE_OTA_H_
+#define FYDEOS_TOGGLE_OTA_H_
+
+namespace fydeos {
+  bool OTAEnabled();
+}
+
+#endif // ifndef FYDEOS_TOGGLE_OTA_H_
Index: update_engine/cros/omaha_request_action.cc
===================================================================
--- update_engine.orig/cros/omaha_request_action.cc
+++ update_engine/cros/omaha_request_action.cc
@@ -43,6 +43,9 @@
 #include "update_engine/cros/payload_state_interface.h"
 #include "update_engine/cros/update_attempter.h"
 #include "update_engine/metrics_utils.h"
+// ---***FYDEOS BEGIN***---
+#include "update_engine/cros/fydeos_toggle_ota.h"
+// ---***FYDEOS END***---
 
 using base::Time;
 using base::TimeDelta;
@@ -1320,6 +1323,11 @@ bool OmahaRequestAction::ShouldIgnoreUpd
     LOG(ERROR) << "All packages were excluded.";
   }
 
+  if (!fydeos::OTAEnabled()) {
+    LOG(INFO) << "fydeos ota disabled, ignore update";
+    return true;
+  }
+
   // Note: We could technically delete the UpdateFirstSeenAt state when we
   // return true. If we do, it'll mean a device has to restart the
   // UpdateFirstSeenAt and thus help scattering take effect when the AU is
Index: update_engine/cros/omaha_request_builder_xml.cc
===================================================================
--- update_engine.orig/cros/omaha_request_builder_xml.cc
+++ update_engine/cros/omaha_request_builder_xml.cc
@@ -24,6 +24,9 @@
 #include "update_engine/common/utils.h"
 #include "update_engine/cros/omaha_request_params.h"
 #include "update_engine/cros/update_attempter.h"
+// ---***FYDEOS BEGIN***---
+#include "update_engine/cros/fydeos_license_checker.h"
+// ---***FYDEOS END***---
 
 using std::string;
 
@@ -290,6 +293,11 @@ string OmahaRequestBuilderXml::GetApp(co
     app_versions = "version=\"" +
                    XmlEncodeWithDefault(app_data.version, kNoVersion) + "\" ";
   }
+  // ---***FYDEOS BEGIN***---
+  string license_id = fydeos::FydeLicenseChecker::Get().GetLicenseId();
+  string fydeos_license_id =
+      "fydeos_license_id=\"" + XmlEncodeWithDefault(license_id) + "\" ";
+  // ---***FYDEOS END***---
 
   string download_channel = params->download_channel();
   string app_channels =
@@ -389,6 +397,9 @@ string OmahaRequestBuilderXml::GetApp(co
       "delta_okay=\"" + delta_okay_str + "\" " +
       "extended_okay=\"" + extended_okay_str + "\" " +
       install_date_in_days_str +
+      // ---***FYDEOS BEGIN***---
+      fydeos_license_id +
+      // ---***FYDEOS END***---
       ">\n" +
          app_body +
       "    </app>\n";
Index: update_engine/cros/update_attempter.cc
===================================================================
--- update_engine.orig/cros/update_attempter.cc
+++ update_engine/cros/update_attempter.cc
@@ -61,6 +61,7 @@
 #include "update_engine/update_manager/omaha_request_params_policy.h"
 #include "update_engine/update_manager/update_manager.h"
 #include "update_engine/update_status_utils.h"
+#include "update_engine/cros/fydeos_toggle_ota.h"
 
 using base::FilePath;
 using base::Time;
@@ -1293,6 +1294,12 @@ void UpdateAttempter::OnUpdateScheduled(
       current_update_flags_ = update_flags_;
     }
 
+    if (!fydeos::OTAEnabled() && pm_ == ProcessMode::UPDATE) {
+      LOG(INFO) << "fydeos ota disabled";
+      ResetUpdateStatus();
+      return;
+    }
+
     switch (pm_) {
       case ProcessMode::UPDATE:
       case ProcessMode::INSTALL:
Index: update_engine/update_manager/official_build_check_policy_impl.cc
===================================================================
--- update_engine.orig/update_manager/official_build_check_policy_impl.cc
+++ update_engine/update_manager/official_build_check_policy_impl.cc
@@ -14,6 +14,8 @@ EvalStatus OnlyUpdateOfficialBuildsPolic
     State* state,
     std::string* error,
     PolicyDataInterface* data) const {
+// ---***FYDEOS BEGIN***---
+  /*
   const bool* is_official_build_p =
       ec->GetValue(state->system_provider()->var_is_official_build());
   if (is_official_build_p != nullptr && !(*is_official_build_p)) {
@@ -28,6 +30,8 @@ EvalStatus OnlyUpdateOfficialBuildsPolic
     LOG(INFO) << "Unofficial build, but periodic update check interval "
               << "timeout is defined, so update is not blocked.";
   }
+  */
+// ---***FYDEOS END***---
   return EvalStatus::kContinue;
 }
 
