diff --git a/cros/omaha_request_action.cc b/cros/omaha_request_action.cc
index a97408ba..a7814f3e 100644
--- a/cros/omaha_request_action.cc
+++ b/cros/omaha_request_action.cc
@@ -1262,9 +1262,9 @@ bool OmahaRequestAction::ShouldIgnoreUpdate(ErrorCode* error) const {
     }
   }
 
-  if (!CheckForRepeatedFpValues(error)) {
-    return true;
-  }
+  // if (!CheckForRepeatedFpValues(error)) {
+  //   return true;
+  // }
 
   if (hardware->IsOOBEEnabled() && !hardware->IsOOBEComplete(nullptr) &&
       (response_.deadline.empty() ||
