Index: update_engine/cros/omaha_request_action.cc
===================================================================
--- update_engine.orig/cros/omaha_request_action.cc
+++ update_engine/cros/omaha_request_action.cc
@@ -1282,9 +1282,9 @@ bool OmahaRequestAction::ShouldIgnoreUpd
     }
   }
 
-  if (!CheckForRepeatedFpValues(error)) {
-    return true;
-  }
+  // if (!CheckForRepeatedFpValues(error)) {
+  //   return true;
+  // }
 
   if (hardware->IsOOBEEnabled() && !hardware->IsOOBEComplete(nullptr) &&
       (response_.deadline.empty() ||
