Index: update_engine/cros/hardware_chromeos.cc
===================================================================
--- update_engine.orig/cros/hardware_chromeos.cc
+++ update_engine/cros/hardware_chromeos.cc
@@ -149,7 +149,7 @@ void HardwareChromeOS::Init() {
 }
 
 bool HardwareChromeOS::IsOfficialBuild() const {
-  return VbGetSystemPropertyInt("debug_build") == 0;
+  return false;
 }
 
 bool HardwareChromeOS::IsNormalBootMode() const {
