diff --git a/cros/boot_control_chromeos.cc b/cros/boot_control_chromeos.cc
index 17659ae7..a84bc481 100644
--- a/cros/boot_control_chromeos.cc
+++ b/cros/boot_control_chromeos.cc
@@ -110,8 +110,8 @@ bool BootControlChromeOS::Init() {
   // devices, so we will pretend we have only one slot in that case.
   if (IsRemovableDevice(boot_disk_name_)) {
     LOG(INFO)
-        << "Booted from a removable device, pretending we have only one slot.";
-    num_slots_ = 1;
+        << "Booted from a removable device, but still pretending we have two slots.";
+    num_slots_ = 2;
   } else {
     // TODO(deymo): Look at the actual number of slots reported in the GPT.
     num_slots_ = 2;
