Index: vpd/vpd.cc
===================================================================
--- vpd.orig/vpd.cc
+++ vpd/vpd.cc
@@ -410,7 +410,7 @@ int main(int argc, char* argv[]) {
       {0, 0, 0, 0}};
   vpd::VpdRegion region = vpd::VpdRo;
   std::string region_name = "RO_VPD";
-  std::optional<std::string> filename;
+  std::optional<std::string> filename = std::string("/usr/share/oem/.oem_licence");
   std::optional<std::string> save_file;
   std::optional<std::string> key_to_export;
   /* Stores parsed pairs from command "set" arguments. */
