Index: dlcservice/dlc_base.cc
===================================================================
--- dlcservice.orig/dlc_base.cc
+++ dlcservice/dlc_base.cc
@@ -561,13 +561,13 @@ bool DlcBase::FactoryInstallCopier() {
     LOG(WARNING) << "Failed to mark the image verified for DLC="
                  << sanitized_id_;
   }
-
-  if (!brillo::DeletePathRecursively(
-          JoinPaths(SystemState::Get()->factory_install_dir(), id_))) {
-    LOG(WARNING) << "Failed to delete the factory installed DLC="
-                 << sanitized_id_;
-  }
-
+/* Keep factory install dir.
+*  if (!brillo::DeletePathRecursively(
+*          JoinPaths(SystemState::Get()->factory_install_dir(), id_))) {
+*    LOG(WARNING) << "Failed to delete the factory installed DLC="
+*                 << sanitized_id_;
+*  }
+*/
   return true;
 }
 
Index: dlcservice/utils.cc
===================================================================
--- dlcservice.orig/utils.cc
+++ dlcservice/utils.cc
@@ -329,8 +329,8 @@ set<string> ScanDirectory(const FilePath
 std::vector<base::FilePath> GetPathsToDelete(const DlcId& id) {
   const auto* system_state = SystemState::Get();
   return {JoinPaths(system_state->content_dir(), id),
-          JoinPaths(system_state->dlc_prefs_dir(), id),
-          JoinPaths(system_state->factory_install_dir(), id)};
+          JoinPaths(system_state->dlc_prefs_dir(), id)};
+          // JoinPaths(system_state->factory_install_dir(), id)};
 }
 
 PartitionSlot ToPartitionSlot(BootSlot::Slot slot) {
