diff --git a/init/upstart/cgroups.conf b/init/upstart/cgroups.conf
index 8f6b68f16a..79ddae54e8 100644
--- a/init/upstart/cgroups.conf
+++ b/init/upstart/cgroups.conf
@@ -34,6 +34,9 @@ script
     fi
     mount -t cgroup cgroup /sys/fs/cgroup/${subsys} \
           -o ${subsys},${mount_opts} || log ${subsys}
+    if [ "${subsys}" = "memory" ]; then
+      echo 1 > /sys/fs/cgroup/memory/memory.use_hierarchy
+    fi
     mkdir -p /sys/fs/cgroup/${subsys}/session_manager_containers
     if id -u android-root 2>/dev/null; then
       chown android-root:android-root \
