diff --git a/init/clobber/clobber_state.cc b/init/clobber/clobber_state.cc
index b2632a27fb..6e245c3d5c 100644
--- init/clobber/clobber_state.cc
+++ init/clobber/clobber_state.cc
@@ -509,7 +509,7 @@ std::vector<base::FilePath> ClobberState::GetPreservedFilesList() {
     preserved_files.push_back(base::FilePath(path));
   }
 
-  if (args_.factory_wipe) {
+  if (args_.factory_wipe || args_.safe_wipe) {
     base::FileEnumerator crx_enumerator(
         stateful_.Append("unencrypted/import_extensions/extensions"), false,
         base::FileEnumerator::FileType::FILES, "*.crx");
diff --git a/init/startup/stateful_mount.cc b/init/startup/stateful_mount.cc
index fed2e2c88a..29dee61a7c 100644
--- init/startup/stateful_mount.cc
+++ init/startup/stateful_mount.cc
@@ -564,6 +564,7 @@ bool StatefulMount::DevUpdateStatefulPartition(
         stateful_.Append(kDevModeFile),
         stateful_.Append("encrypted.block"),
         stateful_.Append("encrypted.key"),
+        stateful_.Append("unencrypted/dlc-factory-images"),
         stateful_dev_image,
         var_target,
         preserve_dir};
