--- a/init/upstart/cgroups.conf
+++ b/init/upstart/cgroups.conf
@@ -112,13 +112,15 @@
 
   # Create a cpu cgroup for each VM type. Default cpu.shares is 1024. Limit the
   # VMs to 7/8ths of that initially.
-  cgroup_dir="/sys/fs/cgroup/cpu"
+  cgroup_dir="/sys/fs/cgroup/cpu/vms"
+if [ -d "${cgroup_dir}" ]; then
   for vm_type in termina plugin ; do
     mkdir -p "${cgroup_dir}/${vm_type}" \
         "${cgroup_dir}/${vm_type}-vcpus" # croslint: disable
     echo 896 > "${cgroup_dir}/${vm_type}/cpu.shares"
     echo 896 > "${cgroup_dir}/${vm_type}-vcpus/cpu.shares"
   done
+fi
 
   schedtune_root_dir="/sys/fs/cgroup/schedtune"
   if [ -d "${schedtune_root_dir}" ]; then
