diff --git a/init/BUILD.gn b/init/BUILD.gn
index 4ebf1f100e..6ab14462b8 100644
--- a/init/BUILD.gn
+++ b/init/BUILD.gn
@@ -40,7 +40,7 @@ pkg_config("target_defaults") {
     "USE_LVM_STATEFUL_PARTITION=${use.lvm_stateful_partition}",
     "USE_PRJQUOTA=${use.prjquota}",
     "USE_TPM2=${use.tpm2}",
-    "USE_TPM_INSECURE_FALLBACK=${use.tpm_insecure_fallback}",
+    "USE_TPM_INSECURE_FALLBACK=1",
   ]
   pkg_deps = [
     "libbrillo",
