Index: init/startup/chromeos_startup.cc
===================================================================
--- init.orig/startup/chromeos_startup.cc
+++ init/startup/chromeos_startup.cc
@@ -851,7 +851,8 @@ int ChromeosStartup::Run() {
 
   base::FilePath pcr_extend_failed =
       stateful_.Append(kVersionPCRExtendFailedFile);
-  if (!ExtendPCRForVersionAttestation()) {
+  const bool presume_success = true;
+  if (!presume_success && !ExtendPCRForVersionAttestation()) {
     // At the moment we'll only log it but not force reboot or recovery.
     // TODO(b/278071784): Monitor if the failure occurs frequently and later
     // change this to reboot/send to recovery when it failed.
